src/plugins/intel_cpu/src/nodes/mkldnn_gather_node.cpp

```diff
@@ -151,6 +158,12 @@ void MKLDNNGatherNode::prepareParams() {
         afterAxisSizeInBytes = afterAxisSize * dataTypeSize;
         axisAndAfterAxisSizeInBytes = axisDim * afterAxisSizeInBytes;
         srcAfterBatchSizeInBytes = betweenBatchAndAxisSize * axisAndAfterAxisSizeInBytes;
+
+        const auto selectedPD = getSelectedPrimitiveDescriptor();
+        if ((x64::mayiuse(x64::avx512_common) || x64::mayiuse(x64::avx2)) && afterAxisSize == 1)
+            selectedPD->setImplementationType(jit);
+        else
+            selectedPD->setImplementationType(ref_any);
```

Comment:
Could you please elaborate why we change the implementation type for the already selected PD? Typically we set implantation types for PD at the `initSupportedPrimitiveDescriptors` stage since they are used at the optimal PD selection stage.

Reply:
Oh, I see.